### ISSUE TITLE
about  inference  after  train  our  datasets 

### ISSUE THREAD

**Issue by QzYER** [mlfoundations/open_clip#602]
Thank you for the author's open-source work. May I ask if the model can be loaded and the code can be inferred after open-source fine-tuning? The official reasoning demo only includes some official weights. How should the model be inferred after fine-tuning?

**Comment by gabrielilharco** [comment]
You can use a local path to a checkpoint when loading a model. E.g. `open_clip.create_model_and_transforms('ViT-B-32', pretrained='/your/path/to/checkpoint.pt')`

**Comment by QzYER** [comment]
Thank you for your prompt response. After fine-tuning my dataset using the Wise FT open source project, I followed your instructions to infer my weight and found it impossible to infer.

**Comment by rom1504** [comment]
impossible in what way?

**Comment by QzYER** [comment]
> impossible in what way?
@rom1504 
This is a model that was fine-tuned using Wise FT open-source. I only fine-tuned one epoch,
The mod
![9e8a247cbd6c105d1f7a47f1711a2c3e](https://github.com/mlfoundations/open_clip/assets/32487883/d41ea063-cd26-4aa1-ab48-c3c1d6dd8a87)
el after fine-tuning is as follows：

 and the inference code is as follows:
import torch
from PIL import Image
import open_clip

model, _, preprocess = open_clip.create_model_and_transforms('ViT-B-32', pretrained='./wise-ft/model/finetuned/checkpoint_1.pt')
tokenizer = open_clip.get_tokenizer('ViT-B-32')

image = preprocess(Image.open("/home/bowen/Downloads/wise-ft/images/imagenet/val/n01498041/0.000116_digital clock _ digital clock_0.865662.jpg")).unsqueeze(0)
text = tokenizer(["a diagram", "a dog", "a cat"])

with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)

print("Label probs:", text_probs)  # prints: [[1., 0., 0.]]



The operation log is as follows:
![f5f68cbc68f600fce8ac2fdf3c143555](https://github.com/mlfoundations/open_clip/assets/32487883/c24a0fa0-66c0-43ad-83e6-04d9a46f56e2)

**Comment by gabrielilharco** [comment]
Hey @QzYER, this is because the checkpoints from the wise-ft codebase are not directly compatible with checkpoints from this codebase.

In wise-ft we discard the text encoder from CLIP and add new parameters (a linear layer on top of image encoder) to make it an image classifier. When fine-tuning using that codebase, the new parameters are trainable.

If you're looking for other codebases where we don't introduce new weights when fine-tuning, we also have https://github.com/mlfoundations/task_vectors and https://github.com/mlfoundations/patching. Even on those codebases, the checkpoints are not directly compatible with this codebase, although it would be possible to make it so with some checkpoint hacking.

**Comment by QzYER** [comment]
> 
@gabrielilharco 
Thank you for patiently answering my questions!
